src/intervals/conversion.jl

```diff
@@ -116,6 +116,21 @@ function atomic(::Type{Interval{T}}, x::S) where {T<:AbstractFloat, S<:AbstractF
     return atomic(Interval{T}, xrat)
 end
 
+function atomic(::Type{Interval{BigFloat}}, x::S) where {S<:AbstractFloat}
```

Comment:
There's no need to copy and paste this code (that is almost always a bad idea). 
The only thing that needs doing is to make a function `_rationalize` that knows what type of `Float` is passed in and calls `rationalize` with the corresponding integer type.

Reply:
Agreed with being almost always a bad idea copying/pasting code like that.

I understood your last reply (of using dispatch) as applying to `atomic()` rather than to `rationalize()`. The originally intended one is now abundantly clear.

The `_rationalize()` method requires that _some_ calling methods to be changed from `rationalize(...)` to `_rationalize(...)`.

Here's the list (on master's) of the sources that currently call `rationalize()`:

```bash
IntervalArithmetic.jl$ grep rationalize $(find . -name '*jl')
./test/interval_tests/construction.jl:    #@test Interval{Rational{Int}}(pi) == Interval(rationalize(1.0*pi))
./test/interval_tests/construction.jl:        @test Interval(1//10).lo == rationalize(0.1)
./src/decorations/intervals.jl:#     convert(DecoratedInterval{T}, rationalize(x))
./src/intervals/conversion.jl:    xrat = rationalize(x)
./src/intervals/conversion.jl:    Interval(rationalize(T, x))
./src/intervals/conversion.jl:    Interval(rationalize(T, x))
```
I imagine that the tests should be left alone, as in them, `rationalize()` is not being used in Interval construction and calls are part of RHS's — hence, expected values.

The cases in `conversion.jl` are pretty clear.

What about the one in `./src/decorations/intervals.jl`? Should be left untouched or changed to `_rationalize()`? (I'm unfamiliar with that part of the code and it's functionality).

Best regards!

Reply:
Oh, in the case that `rationalize()` calls on _multiple_ sources are to be changed, in which source file would be best to define `_rationalize()` in your estimate?